### ISSUE TITLE
I confused by the identity_reprojection_losses. please reply me when you are convenient, thanks very much!

### ISSUE THREAD

**Issue by Suodislie** [nianticlabs/monodepth2#301]
pred = outputs[("color", frame_id, scale)]
reprojection_losses.append(self.compute_reprojection_loss(pred, target))
...
pred = inputs[("color", frame_id, source_scale)]
identity_reprojection_losses.append( self.compute_reprojection_loss(pred, target))
...
combined = torch.cat((identity_reprojection_loss, reprojection_loss), dim=1)
to_optimxs = torch.min(combined, dim=1)

I don't understand the above steps, what is the meaning of identity_reprojection_losses(input_t0,input_t1), and why concat it with reprojection_losses and then  take the minimum value of combined ?
Please reply me when you are convenient, Thanks very much!

**Comment by mrharicot** [comment]
Hi,

The concat is here so that we can take the minimum for each pixel between the depth reprojection and the identity reprojection.
We concatenate along the second dimension `dim=1` and then take the minimum on that same dimension.

I hope this helps!
